orchagent/switchorch.cpp

```diff
@@ -473,24 +476,197 @@ void SwitchOrch::doAppSwitchTableTask(Consumer &consumer)
     }
 }
 
+bool SwitchOrch::setSwitchHashFieldListSai(const SwitchHash &hash, bool isEcmpHash) const
+{
+    const auto &oid = isEcmpHash ? m_switchHashDefaults.ecmpHash.oid : m_switchHashDefaults.lagHash.oid;
+    const auto &hfSet = isEcmpHash ? hash.ecmp_hash.value : hash.lag_hash.value;
+
+    std::vector<sai_int32_t> hfList;
+    std::transform(
+        hfSet.cbegin(), hfSet.cend(), std::back_inserter(hfList),
+        [](sai_native_hash_field_t value) { return static_cast<sai_int32_t>(value); }
+    );
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_HASH_ATTR_NATIVE_HASH_FIELD_LIST;
```

Comment:
Do you have plan to support `SAI_SWITCH_ATTR_ECMP_HASH_IPV4`/`SAI_SWITCH_ATTR_ECMP_HASH_IPV4_IN_IPV4`/`SAI_SWITCH_ATTR_ECMP_HASH_IPV6` /`SAI_SWITCH_ATTR_LAG_HASH_IPV4`/`SAI_SWITCH_ATTR_LAG_HASH_IPV4_IN_IPV4`/`SAI_SWITCH_ATTR_LAG_HASH_IPV6`  besides the default hash field list?

Reply:
@baxia-lan no. Implementation is according to the HLD. Also, this topic was discussed during Community design review and folks agreed that it requires some HLD/SAI extension in order to support hashing for different types of encapsulated frames per ASIC vendor. In any case, currently no plans to work on it.

Reply:
Thanks for the heads up. Could you please provide the pointer of the HLD discussion recording for these concerns? Also if possible, could you please help me to clarify the SAI extension needed?